### PR TITLE
Add support for EXT-X-MAP and EXT-X-BYTERANGE

### PR DESCRIPTION
**Pull request by carlanton**
Fixes some of the byte range issues discussed in #47 + support for EXT-X-MAP.

### CODE REVIEW

File: src/main/java/com/iheartradio/m3u8/data/ByteRange.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package com.iheartradio.m3u8.data;
+
+import java.util.Objects;
+
+public class ByteRange {
+    private final long mSubRangeLength;
+    private final Long mOffset;
```
Comment (Wopple) [inline src/main/java/com/iheartradio/m3u8/data/ByteRange.java]: Please use a primitive, see `IFrameStreamInfo.getAverageBandwidth` as an example.

File: src/main/java/com/iheartradio/m3u8/data/ByteRange.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return "ByteRange{" +
+                "mSubRangeLength=" + mSubRangeLength +
+                ", mOffset=" + mOffset +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (Wopple) [inline src/main/java/com/iheartradio/m3u8/data/ByteRange.java]: Though I don't think it makes a functional difference, the other data classes have been using `instanceof` for the type check.

File: src/main/java/com/iheartradio/m3u8/data/ByteRange.java (hunk 1 above)
Comment (Wopple) [reply src/main/java/com/iheartradio/m3u8/data/ByteRange.java]: Actually, there is no `null` case for the offset. I'll refer to my comment here:

https://github.com/iheartradio/open-m3u8/pull/47#discussion_r130231079

**Comment by Wopple** [comment]
All in all, nice work! The offset is a little nuanced and difficult to handle, but once that gets updated this should be good to merge.

**Comment by aaslami** [comment]
When will this be released? I see that last release was 0.2.6 (Jan 2017). Thanks.

**Comment by yngwietiger** [comment]
When will this be released? I see that last release was 0.2.6 (Jan 2017). Thanks.

Or should I switch to the carlanton/m3u8-parser?

**Pull request merged**